Copy Checked C headers to clang system header directory.

dtarditi opened the pull request:
The Checked C header files are stored in the Checked C repo.   We want to copy them to the clang system header directory.  The Checked C repo is stored under llvm\projects\checked-wrapper, not clang.  This change causes the cmake files for LLVM to pick up the cmake file for the Checked C header files.  That adds a new target that can be used by the clang build system.

dtarditi committed:
Copy Checked C headers to clang system header directory.

The Checked C header files are stored in the Checked C repo.  This
adds a new dependency  to the LLVM build system that gets used by the clang
build system.  The dependency is used by the clang cmake files.

dtarditi committed:
Set variable needed by clang build system.

lenary reviewed (approved):
LGTM :+1:

lenary reviewed (changes requested):
I've just sent you a patch for this branch, once it's applied I'll approve this, as it should then scope "CHECKEDC_IN_TREE" correctly.

dtarditi committed:
Make CHECKEDC_IN_TREE globally defined.

Early versions of CMake used to make variables globally scoped.  Now
CMake uses dynamically-scoped variables, where a new scope is created at
the directory level and the variables are copied into the scope from the
parent scope.  This meant that setting the variable in nested CMake file
had no global effect.

Move setting the variable to the root CMake file for LLVM, so that the value
is globally visible.

This patch was contributed by Sam Elliott.

lenary reviewed (approved):
Great!

dtarditi commented:
Thanks for fix!   I hadn't realized that CMake uses dynamically-scoped variables with an unusual semantics for scoping.

lenary commented:
Yeah, it was a guess I had that I looked up in the cmake docs to see if it was the issue.
